### ISSUE TITLE
Bad results for multimodal similarity search

### ISSUE THREAD

**Issue by mct10** [facebookresearch/fairseq#5265]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
Following the evaluation of speech encoder in [SpeechMatrix paper](https://arxiv.org/abs/2211.04508), I ran a multimodal similarity search on [FLEURS ](https://huggingface.co/datasets/google/fleurs) validation set between English text embeddings and English speech embeddings, but the error rates were **100%**.

To be specific, the speech embeddings were generated following [this example](https://github.com/facebookresearch/fairseq/blob/ust/examples/speech_matrix/speech_laser_encoders.md#speechlaser-encoders), and the text embeddings were generated using [the script in LASER](https://github.com/facebookresearch/LASER/blob/main/tasks/embed/embed.sh) while the encoders are replaced with LASER.

The similarity search result was calculated basically with [this script](https://github.com/facebookresearch/LASER/blob/main/source/xsim.py).

I am wondering whether I have generated the embeddings correctly, or are there some processing required for the multimodal embeddings when running the similarity search?

#### What have you tried?
(1) Similarity search between _English text_ embeddings and _German text_ embeddings, the error rate was 0%.
(2) Similarity search between _English speech_ embeddings and _German speech_ embeddings, the error rate was 0%.

These two may confirm that I have generated the embeddings correctly.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or main):
 - PyTorch Version (e.g., 1.0): 
 - OS (e.g., Linux):
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:
